src/resultsdisassemblypage.cpp

```diff
@@ -123,6 +136,39 @@ ResultsDisassemblyPage::ResultsDisassemblyPage(QWidget* parent)
             }
         }
     });
+
+#if KF5SyntaxHighlighting_FOUND
+    QStringList schemes;
+
+    for (const auto& entry : m_repository->definitions()) {
+        schemes.push_back(entry.name());
+    }
+
+    auto connectCompletion = [schemes, this](QComboBox* box, auto* model) {
+        auto m = new QStringListModel(this);
+        m->setStringList(schemes);
+
+        auto completer = new QCompleter(this);
+        completer->setModel(m);
+        completer->setCaseSensitivity(Qt::CaseInsensitive);
+        completer->setCompletionMode(QCompleter::PopupCompletion);
+        box->setCompleter(completer);
+
+        connect(completer, qOverload<const QModelIndex&>(&QCompleter::activated), this,
```

Comment:
why is this using a completer? what happens when you select something with the mouse graphically?

why is the combo box even editable, i.e. why the completer? what happens if one enters garbage?

Reply:
> why is this using a completer?

To suggest highlighting definitions. Since there are multiple ways for some languages (C++, CPP, CXX, cpp, ...) I use a complete to suggest an installed definition.

> what happens when you select something with the mouse graphically?

it works (`QComboBox::activated` triggers)

> why is the combo box even editable, i.e. why the completer?

Its the easiest way to add a search function. I didn't want a list, since there 350 different definitions

> what happens if one enters garbage?

Syntax Highlighting gets disabled

Reply:
put differently, do we need both? isn't `QComboBox::activated` getting emitted when we select an item from the completer? do we really need both? if so, share the common code in a lambda and call it from both slots to make it clear that these do the same thing

we also need the back-sync code to select the current syntax highlighting mode from the model in the ui

Reply:
`QCompleter::activated` is not needed (qt docs weren't clear)